### DISCUSSION TITLE
[C++] How to set number of threads using openmp

### DISCUSSION THREAD

**Discussion by bitnick10** [microsoft/onnxruntime#5628] [Other Q&A] [answered]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):win10
- ONNX Runtime installed from (source or binary):nuget
- ONNX Runtime version:1.5.2

`session_options_.SetIntraOpNumThreads(1);`
WARNING: Since openmp is enabled in this build, this API cannot be used to configure intra op num threads. Please use the openmp environment variables to control the number of threads.

**Comment by zhuxiaoxuhit** [comment]
add this cmd in ur ~/.bashrc 
```
export OMP_NUM_THREADS=8
```

**Comment by zhuxiaoxuhit** [comment, marked as answer]
OK, ur sysmtem is windows
```
set OMP_NUM_THREADS[=num]
```

**Comment by zhuxiaoxuhit** [comment]
This is a tutorial. 
https://docs.microsoft.com/en-us/cpp/parallel/openmp/reference/openmp-environment-variables?view=msvc-160
